dataflux_pytorch/benchmark/checkpointing/simulated/README.md

```diff
@@ -0,0 +1,115 @@
+
+  * `LAYER_SIZE`: The size of each layer.
+
+2. Set the optional environment variables, if desired:
+  
+  * `PADDING_SIZE`: The number of dummy tensors to add to the state_dict in order to produce checkpoints of desired size.
+    *   Both padding_size and layer_size will impact the size of the checkpoint. Therefore, set layer_size to an appropriate value first, and then adjust padding_size until the checkpoint of the desired size is generated. The default value for padding_size is 4000.
+  
+  * `SAMPLE_COUNT`: The number of times save_checkpoint/load_checkpoint is called. The default is 8.
+
+  * `USE_FSSPEC`: If set to true, the code will use [gcsfs/fsspec](https://github.com/fsspec/gcsfs) in order to save_checkpoint/restore_checkpoint from GCS.
+    *   If not set, it will use GCS Connector for Pytorch by default.
+
+### Installing Requirements:
+ 
+* Install the required dependencies using the following commands:
+
+```shell
+ pip install -r dataflux_pytorch/benchmark/requirements.txt; 
+ pip install .
+```
+### GCS Auth
+
+Run the following commands to authenticate your application so it can access GCS buckets.
+```shell
+gcloud config set project PROJECT_ID
+gcloud auth application-default login
+```
+
+### Running
+
+_Note: The following instructions assume that you have Jobset and Kueue enabled on your GKE cluster. For easier compatibility, we recommend creating a cluster using [XPK](https://github.com/google/xpk) which will configure these features automatically._
+
+1.  To connect to GKE, you need to generate a kubeconfig file. This can be done using the following command:
+
+```shell
+gcloud container clusters get-credentials {YOUR-GKE-CLUSTER-NAME} --zone {ZONE} --project {YOUR-GCP-PROJECT}
+```
+
+2.  Build the container.
+
+Ensure that your working directory is `dataflux-pytorch`
+```shell
+docker build -t {YOUR_CONTAINER_NAME} .
+```
+
+3.  Upload your container to the container registry:
+
+```shell
+docker tag {YOUR_CONTAINER_NAME} gcr.io/{YOUR-GCP-PROJECT}/{YOUR_CONTAINER_NAME}
+docker push gcr.io/{YOUR-GCP-PROJECT}/{YOUR_CONTAINER_NAME}
+```
+
+4.  Update the values in `benchmark-deployment.yaml`.
+
+5.  Deploy the configuration using the following command:
+
+```shell
+kubectl apply -f /app/dataflux_pytorch/benchmark/checkpointing/simulated/benchmark-deploy.yaml
+```
+
+
+### Benchmarking Results.
+
+The table below contains benchmarking times for saving checkpoints to GCS. The average save/load time is taken over 8 calls to  save_checkpoint and load_checkpoint. The tests were conducted on a single GKE cluster containing several nodes, with each VM having the `n2-standard-32` configuration. The GKE cluster, and the GCS bucket were located in `us-central1`.
+
+### Checkpoint Save & Load times.
+
+
+| Num_nodes | Parameters | Checkpoint Type | Num_layers + padding_size | Size of each shard (in GB) | Total size (in GB) | Save time (in seconds) | Load time (in seconds) | Average Throughput (in GBps) |
```

Comment:
The average throughput column doesn't really make sense here because it's not clear if it's referring to save or load. You may want to separate this out to 2 different tables for save/load like we've done in some of the other docs.

Reply:
Oh I forgot the egress column, I think splitting it into 2 will be better fir readability. 

Done